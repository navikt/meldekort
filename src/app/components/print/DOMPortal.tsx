import * as React from 'react';
import { ClassAttributes } from 'react';
import { createPortal } from 'react-dom';

class DOMPortal extends React.Component<ClassAttributes<any>, {}> {
  render() {
    return createPortal(this.props.children, document.body);
  }
}

export default DOMPortal;
