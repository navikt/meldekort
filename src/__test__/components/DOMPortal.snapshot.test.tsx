import * as React from 'react';
import * as renderer from 'react-test-renderer';
import ProviderWrapper from '../testSetup/providerWrapper';
import DOMPortal from '../../app/components/print/DOMPortal';

it('DOMPortal', () => {
  const tree = renderer.create(
    <ProviderWrapper>
      <DOMPortal />
    </ProviderWrapper>
  );
  expect(tree).toMatchSnapshot();
});
