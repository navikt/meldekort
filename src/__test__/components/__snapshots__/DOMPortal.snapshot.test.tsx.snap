// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOMPortal 1`] = `null`;
